itou/www/apply/tests/tests_process.py

```diff
@@ -281,6 +286,61 @@ def test_accept(self, *args, **kwargs):
         self.assertFormError(response, "form_user_address", "city", "Ce champ est obligatoire.")
         self.assertFormError(response, "form_user_address", "post_code", "Ce champ est obligatoire.")
 
+    def test_accept_without_hiring_end_at(self, *args, **kwargs):
+        """Test the `accept` transition."""
+        create_test_cities(["54", "57"], num_per_department=2)
+        city = City.objects.first()
+        today = timezone.localdate()
+
+        job_seeker = JobSeekerWithAddressFactory(city=city.name)
+        address = {
+            "address_line_1": job_seeker.address_line_1,
+            "post_code": job_seeker.post_code,
+            "city": city.name,
+            "city_slug": city.slug,
+        }
+        siae = SiaeWithMembershipFactory()
+        siae_user = siae.members.first()
+
+        for state in JobApplicationWorkflow.CAN_BE_ACCEPTED_STATES:
```

Comment:
hmm, les `for` dans un test sont en général à éviter, essayer d'utiliser `subTest` https://www.caktusgroup.com/blog/2017/05/29/subtests-are-best/

Comme ça, on voit pour quelle valeur de `state` le test échoue sans devoir debugger à la main !

Reply:
Je viens d'apprendre un truc ! Merci !

Reply:
Ajout de subtest, et intégration dans le test `test_accept`, en utilisant une liste de tuples (hiring_end_at, states).